packages/components/src/Modal/Modal.tsx

```diff
@@ -0,0 +1,126 @@
+import React from 'react';
+import styled from 'styled-components';
+import { MaterialClear } from '@t3n/icons';
+import { theme, ThemeProps } from '@t3n/theme';
+import { Card } from '../Card';
+import { Box } from '../Box';
+import { Heading } from '../Heading';
+import { Button } from '../Button';
+
+export interface ModalProps {
+  headline: string;
+  onClose: () => void;
+  onConfirm: () => void;
+  buttonLabel: string;
+}
+
+const ModalWrapper = styled.div`
+  position: fixed;
+  top: 0;
+  left: 0;
+  width: 100%;
+  height: 100%;
+  z-index: 200;
+  align-items: center;
+  display: flex;
+  justify-content: center;
+
+  @media screen and (max-width: ${(props: ThemeProps) =>
+      props.theme.breakpoints[0]}) {
+    align-items: flex-end;
+  }
+`;
+
+const ModalBackdrop = styled.div`
+  position: absolute;
+  top: 0;
+  left: 0;
+  width: 100%;
+  height: 100%;
+  background: rgba(0, 0, 0, 0.6);
+`;
+
+const StyledModal = styled(Card)`
+  position: relative;
+  overflow: inherit;
+  z-index: 1;
+  max-height: 100%;
+  overflow: auto;
+  margin-bottom: 0;
+`;
+
+const StyledHeading = styled(Heading)`
+  width: 85%;
+`;
+
+const StyledIconContainer = styled(Box)`
+  position: absolute;
+  top: 10px;
+  right: 10px;
+  border-radius: 50%;
+  fill: ${theme.colors.text.primary};
+
+  &:hover {
+    cursor: pointer;
+    background: ${theme.colors.brand.red};
+    fill: ${theme.colors.text.inverse};
+  }
+`;
+
+const StyledButtonBox = styled(Box)`
+  @media screen and (max-width: ${(props: ThemeProps) =>
+      props.theme.breakpoints[0]}) {
+    flex-direction: column;
+  }
+`;
+
+const CloseIcon: React.FC<{ onClick: () => void }> = ({ onClick }) => {
+  return (
+    <StyledIconContainer
+      onClick={onClick}
+      alignItems="center"
+      justifyContent="center"
+      bg="shades.grey232"
+      p={1}
+      display="flex"
+      width="40px"
+      height="40px"
+    >
+      <MaterialClear width="24px" height="24px" />
+    </StyledIconContainer>
+  );
+};
+
+export const Modal: React.FC<ModalProps> = ({
+  headline,
+  onConfirm,
+  onClose,
+  buttonLabel,
+  children,
+}) => {
+  return (
+    <ModalWrapper>
+      <StyledModal elevate width={[1, 2 / 3, 2 / 5]}>
+        <CloseIcon onClick={onClose} />
+        <StyledHeading styleAs="h4" as="h4" mt={1} mb={3}>
+          {headline}
+        </StyledHeading>
+        {children}
+        <StyledButtonBox display="flex" justifyContent="flex-end" mt={3}>
+          <Button
```

Comment:
Ich weiss nicht ob wir das wirklich so haben wollen. Die Butons hier sind mir zu fix. Wir werden ganz sicher an den Punkt kommen wo man mal nur einen Button haben will. Oder vor allem gar keinen. Ich plädiere dafür, dass die Buttons über children reinkommen müssen. Was sagst du, @itscursedphil ?

Reply:
So stehts im Ticket und ich habe auch nochmal nachgefragt. Eigentlich muss man, wenn man das so haben will, aber auch nur in der Komponente selbst die beiden Buttons auf nicht required stellen und an dieser Stelle dann eine Abfrage machen. Der Confirm-Button ist von außen steuerbar.

Reply:
Wäre also minimaler Anpassungsaufwand. Würde das dann nicht über Children reingeben.

Reply:
Ah, sry. Habe das Ticket noch nicht angeschaut. Dann wird es ganz sicher bald ein Ticket geben, dass die BUttons rauskommen 😂  Den Case werden wir sicher haben. Sonst mega! Magst du noch Screenshots dranpacken?

Reply:
Ist im Ticket so definiert, das stimmt. Aber wo ich jetzt darüber nachdenke, bin ich mir nicht sicher, ob wir mit dem Modal nur einen Container bereitstellen sollten, dessen gesamter Body per children flexibel befüllt wird.

Reply:
Das wäre mein Vorschlag, ja. Headline kann man gerne lassen. Aber Buttons. müssen eigentlich von außen rein. Nimm mal sowas wie den AdBlocker-Hinweis. Kann man damit nicht machen.

Reply:
<img width="523" alt="Bildschirmfoto 2020-04-07 um 13 43 05" src="https://user-images.githubusercontent.com/57144261/78665236-c8c5e100-78d5-11ea-8646-2ecea356fd06.png">
<img width="485" alt="Bildschirmfoto 2020-04-07 um 13 43 51" src="https://user-images.githubusercontent.com/57144261/78665270-d5e2d000-78d5-11ea-83bb-592defb1049a.png">

Reply:
Ist halt wieder die Frage, wie viel Konsistenz man für Flexibilität opfern will. Aber ja, genau die Überlegung hatte ich auch gerade, und ich denke, dass man trotzdem für beide Fälle per Props eine gute API schaffen kann. Aber vielleicht eher als Ausbau-Stufe betrachten, das wart auch mit Melanie vorhin unser Resumé.